webapp/src/app/home/home.component.ts

```diff
@@ -13,9 +14,21 @@ import { lastValueFrom } from 'rxjs';
 export class HomeComponent {
   leaderboardService = inject(LeaderboardService);
 
+  // timeframe for leaderboard
+  // example: 2024-09-19T10:15:30+01:00
+  protected after: string | undefined = undefined;
+  protected before: string | undefined = undefined;
+
+  constructor() {
+    inject(ActivatedRoute).queryParamMap.subscribe((params) => {
+      this.after = params.get('after')?.replace(' ', '+') ?? undefined;
+      this.before = params.get('before')?.replace(' ', '+') ?? undefined;
+    });
+  }
+
   query = injectQuery(() => ({
     queryKey: ['leaderboard'],
-    queryFn: async () => lastValueFrom(this.leaderboardService.getLeaderboard()),
+    queryFn: async () => lastValueFrom(this.leaderboardService.getLeaderboard(this.before, this.after)),
     gcTime: Infinity
```

Comment:
Just remove this, I like that refetches the data by default after 5min. This keeps it somewhat up-to-date :) Also when you revisit the page after a day or so

Reply:
I think you might be confusing a few elements here.
`gcTime` only determines how often the request is garbage-collected (so for example while the site is inactive in the background).
Even with `gcTime: Infinity`, there are many instances where the query is re-fetched, e.g. window re-focus, site focus change, site refresh, change in query parameters, navigation, etc., so are we sure, that we want to have an additional garbage collection for the site while in the background?

Reply:
I see. But, I mean this basically says you don't want to delete the fetched data after 5 min if we no longer use the query, i.e. the component with `injectQuery` is unmounted, so it is cached forever. 

I think you can just remove it 🤔